src/convertToObject.js

```diff
@@ -7,6 +7,20 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  const x = sourceString
+    .toString()
+    .split(';')
+    .map((item) => item.toString().split(':'));
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. The `toString()` call on `sourceString` is unnecessary since `sourceString` is already expected to be a string.